router/delivery/http/router_handler.go

```diff
@@ -150,39 +148,66 @@ func (a *RouterHandler) GetOptimalQuote(c echo.Context) (err error) {
+// @Param  tokenIn         query  string  false  "String representation of the sdk.Coin denoting the input token for the exact amount in swap method."                       example(1000000uosmo)
+// @Param  tokenOutDenom   query  string  false  "String representing the list of the output token denominations separated by comma for the exact amount in swap method."    example(uion)
+// @Param  tokenOut        query  string  false  "String representation of the sdk.Coin denoting the output token for the exact amount out swap method."                     example(2353uion)
+// @Param  tokenInDenom    query  string  false  "String representing the list of the input token denominations separated by comma for the exact amount out swap method."    example(uosmo)
+// @Param  poolID          query  string  true  "String representing list of the pool ID."                                                                                   example(1100)
 // @Param  applyExponents  query  bool    false  "Boolean flag indicating whether to apply exponents to the spot price. False by default."
 // @Success 200  {object}  domain.Quote  "The computed best route quote"
 // @Router /router/custom-direct-quote [get]
-func (a *RouterHandler) GetDirectCustomQuote(c echo.Context) error {
+func (a *RouterHandler) GetDirectCustomQuote(c echo.Context) (err error) {
 	ctx := c.Request().Context()
 
-	poolIDs, tokenOutDenom, tokenIn, err := getDirectCustomQuoteParameters(c)
-	if err != nil {
-		return c.JSON(http.StatusBadRequest, domain.ResponseError{Message: err.Error()})
-	}
-
-	shouldApplyExponentsStr := c.QueryParam("applyExponents")
-	shouldApplyExponents := false
-	if shouldApplyExponentsStr != "" {
-		shouldApplyExponents, err = strconv.ParseBool(shouldApplyExponentsStr)
+	defer func() {
 		if err != nil {
-			return c.JSON(domain.GetStatusCode(err), domain.ResponseError{Message: err.Error()})
+			// nolint:errcheck // ignore error
+			c.JSON(domain.GetStatusCode(err), domain.ResponseError{Message: err.Error()})
 		}
+
+		// Note: we do not end the span here as it is ended in the middleware.
+	}()
+
+	var req types.GetDirectCustomQuoteRequest
+	if err := UnmarshalRequest(c, &req); err != nil {
+		return err
 	}
 
-	// Quote
-	quote, err := a.RUsecase.GetCustomDirectQuoteMultiPool(ctx, tokenIn, tokenOutDenom, poolIDs)
+	// Validate the request
+	if err := req.Validate(); err != nil {
+		return err
+	}
+
+	var (
+		tokenIn       *sdk.Coin
+		tokenOutDenom []string
+	)
+
+	if req.SwapMethod() == domain.TokenSwapMethodExactIn {
+		tokenIn, tokenOutDenom = req.TokenIn, req.TokenOutDenom
+	} else {
+		tokenIn, tokenOutDenom = req.TokenOut, req.TokenInDenom
+	}
+
+	var quote domain.Quote
+	if req.SwapMethod() == domain.TokenSwapMethodExactIn {
+		quote, err = a.RUsecase.GetCustomDirectQuoteMultiPool(ctx, *tokenIn, tokenOutDenom, req.PoolID)
+	} else {
+		quote, err = a.RUsecase.GetCustomDirectQuoteMultiPoolInGivenOut(ctx, *tokenIn, tokenOutDenom, req.PoolID)
+	}
```

Comment:
Do we have to do this check twice or can we group it under a single if/else?

Reply:
Since I have updated to include `humanDenoms` functionality, I think now grouping to in a single branch would add more readability issues and duplicated code. Please let me know what do you think. :pray: